### PR TITLE
Adding CMake compilation support

### LINKED ISSUE #14: Task 0.1 - Setup (main.cpp + cmake)
- stwórz plik main.cpp
- CMakeLists.txt
- CMakeLists.txt.in
- plik test.cpp zawierjacy jedynie #include "gtest/gtest.h"

### PR DESCRIPTION
**Pull request by Petrus9107**

### CODE REVIEW

File: shm/CMakeLists.txt [hunk 1]
```diff
@@ -17,4 +53,8 @@ set(SOURCES
     ${CMAKE_SOURCE_DIR}/Map.cpp
 )
 
-add_executable(${CMAKE_PROJECT_NAME} ${SOURCES})
\ No newline at end of file
+add_executable(${CMAKE_PROJECT_NAME} ${SOURCES})
+add_executable(${CMAKE_PROJECT_NAME}-ut ${SOURCES} test.cpp)
+
+target_link_libraries(${CMAKE_PROJECT_NAME}-ut gtest_main)
+add_test(NAME shm_test COMMAND ${CMAKE_PROJECT_NAME}-ut)
```
Comment (m-kop) [inline shm/CMakeLists.txt]: new line

File: shm/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+set(SOURCES
+    ${CMAKE_SOURCE_DIR}/main.cpp
+    ${CMAKE_SOURCE_DIR}/Map.cpp
+)
+
+add_executable(${CMAKE_PROJECT_NAME} ${SOURCES})
+add_executable(${CMAKE_PROJECT_NAME}-ut ${SOURCES} test.cpp)
+
+target_link_libraries(${CMAKE_PROJECT_NAME}-ut gtest_main)
+add_test(NAME shm_test COMMAND ${CMAKE_PROJECT_NAME}-ut)
```
Comment (m-kop) [inline shm/CMakeLists.txt]: new line

File: shm/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+add_definitions(-D Wall -Wextra -Werror -pedantic)
+
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+
+##Insert project source files here.##
+set(SOURCES
+    ${CMAKE_SOURCE_DIR}/main.cpp
+    ${CMAKE_SOURCE_DIR}/Map.cpp
```
Comment (m-kop) [inline shm/CMakeLists.txt]: remove these or add main.cpp to the PR

File: shm/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+)
+
+add_executable(${CMAKE_PROJECT_NAME} ${SOURCES})
+add_executable(${CMAKE_PROJECT_NAME}-ut ${SOURCES} test.cpp)
```
Comment (m-kop) [inline shm/CMakeLists.txt:56]: ditto

**Review by m-kop — CHANGES_REQUESTED** [review]

File: shm/CMakeLists.txt (hunk 3 above)
Comment (Petrus9107) [reply shm/CMakeLists.txt]: added main.cpp

File: shm/CMakeLists.txt (hunk 4 above)
Comment (Petrus9107) [reply shm/CMakeLists.txt:56]: added "empty" test.cpp

File: shm/CMakeLists.txt (hunk 2 above)
Comment (Petrus9107) [reply shm/CMakeLists.txt]: done

File: shm/CMakeLists.txt (hunk 1 above)
Comment (Petrus9107) [reply shm/CMakeLists.txt]: done

**Pull request merged**